client/packages/theia-ecore/src/browser/workspace-delete-handler.ts

```diff
@@ -0,0 +1,190 @@
+        const messageContainer = document.createElement("div");
+        messageContainer.textContent = baseText;
+        const list = document.createElement("ul");
+        list.style.listStyleType = "none";
+        for (const uri of uris) {
+            const listItem = document.createElement("li");
+            let content = uri.path.base;
+            content = content.concat(this.additionalEcoreMsg(uri));
+            listItem.textContent = content;
+            list.appendChild(listItem);
+        }
+        messageContainer.appendChild(list);
+        return messageContainer;
+    }
+
+    protected additionalEcoreMsg(resourceUri: URI): string {
+        if (this.isEcoreFile(resourceUri)) {
+            return " (This will also delete the Ecore resource *.enotation in the same directory)";
+        } else if (resourceUri.path.ext === "") {
+            return " (This will also delete any contained Ecore resources *.ecore/*.enotation)";
+        }
+        return "";
+    }
+
+    /**
+     * Perform deletion of a given URI.
+     *
+     * @param resourceUri URI of selected resource.
+     */
+    protected async delete(resourceUri: URI, options: FileDeleteOptions): Promise<void> {
+        try {
+            if (this.isEcoreFile(resourceUri)) {
+                // delete enotation file from file system as well
+                const enotationUri = this.getEnotationFileUri(resourceUri);
+                await Promise.all([
+                    this.closeWithoutSaving(resourceUri),
+                    this.closeWithoutSaving(enotationUri),
+                    this.fileService.delete(resourceUri, options),
+                    this.fileService.delete(enotationUri, options)
+                ]);
+                await this.modelServerClient.deleteEcoreResources(resourceUri.toString());
+            } else if (this.isEnotationFile(resourceUri)) {
+                const ecoreResourceUri = this.getEcoreFileUri(resourceUri);
+                await Promise.all([
+                    this.closeEcoreDiagramWidget(this.shell, ecoreResourceUri),
+                    this.closeWithoutSaving(resourceUri),
+                    this.fileService.delete(resourceUri, options)
+                ]);
+                await this.modelServerClient.deleteEnotationResource(resourceUri.toString());
+            } else {
+                if (await this.isEcoreDirectory(resourceUri)) {
+                    const ecoreResourceUri = await this.getContainedEcoreFileUri(resourceUri);
+                    const enotationUri = this.getEnotationFileUri(ecoreResourceUri!);
+                    await Promise.all([
+                        this.closeEcoreDiagramWidget(this.shell, ecoreResourceUri!),
+                        this.closeWithoutSaving(enotationUri),
+                        this.fileService.delete(resourceUri, options)
+                    ]);
+                    await this.modelServerClient.deleteEcoreResources(ecoreResourceUri!.toString());
```

Comment:
As mentioned in eclipse-emfcloud/emfcloud-modelserver-theia/issues/50, I'm not convinced that Theia should have to notify the Model Server about file system resource changes; the Model Server itself should check the resources. However, since both Theia and the Model Server may have independent access to the same file system, this is a more complex synchronization issue; so we can think about better solutions later on.

Reply:
Thanks for pointing that out, I agree! We also moved the mentioned issue directly to the modelserver component: https://github.com/eclipse-emfcloud/emfcloud-modelserver/issues/115